### PR TITLE
Adds admin API endpoints for product operations

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/products/get-product.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export default async (req, res) => {
+  const { id } = req.params
+
+  const productService = req.scope.resolve("productService")
+  let product = await productService.retrieve(id)
+
+  if (!product) {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/products/get-product.js:7]: Not necessary - when retrieve throws a `NOT_FOUND` error the express error handler will send the correct status code

File: packages/medusa/src/api/routes/admin/products/list-products.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export default async (req, res) => {
+  const productService = req.scope.resolve("productService")
+  let products = await productService.list({})
+
+  if (!products) {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/products/list-products.js:5]: not needed

Comment (srindom) [reply packages/medusa/src/api/routes/admin/products/list-products.js:5]: List will just return an empty array

File: packages/medusa/src/api/routes/admin/products/update-option.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      await productService.updateOption(product._id, optionId, {
+        title: value.title,
+      })
+    }
+
+    if (value.values) {
+      await productService.updateOption(product._id, optionId, {
+        values: value.values,
+      })
+    }
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/products/update-option.js:30]: Don't need two calls here

File: packages/medusa/src/api/routes/admin/products/update-product.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  if (error) {
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
+  }
+
+  const productService = req.scope.resolve("productService")
+  const oldProduct = await productService.retrieve(id)
+  if (!oldProduct) {
+    res.sendStatus(404)
+    return
+  }
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/products/update-product.js:30]: Not needed

File: packages/medusa/src/api/routes/store/products/get-product.js [hunk 5]
```diff
@@ -11,12 +11,23 @@ export default async (req, res) => {
   }
 
   const productService = req.scope.resolve("productService")
-  const product = await productService.retrieve(value)
+  let product = await productService.retrieve(value)
 
   if (!product) {
     res.sendStatus(404)
     return
   }
```
Comment (srindom) [inline packages/medusa/src/api/routes/store/products/get-product.js:20]: Not needed

File: packages/medusa/src/api/routes/store/products/get-product.js [hunk 6]
```diff
@@ -11,12 +11,23 @@ export default async (req, res) => {
...
+  // product = await productService.decorate(product, [
+  //   "title",
+  //   "description",
+  //   "tags",
+  //   "handle",
+  //   "images",
+  //   "options",
+  //   "variants",
+  //   "published",
+  // ])
```
Comment (srindom) [inline packages/medusa/src/api/routes/store/products/get-product.js:30]: Should not be commented

File: packages/medusa/src/api/routes/admin/products/update-option.js (hunk 3 above)
Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: Not sure how to combine this without checking multiple combinations in `if () {} else if` etc..

Comment (srindom) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: What's wrong with `await productService.updateOption(product._id, optionId, value)`?

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: Ah I see... I had my head elsewhere - my bad. (late night sesh)

Comment (srindom) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: Actually when looking at the ProductService it seems that you can only update the title

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: The values are being updated, but we aren't really handling them at all right?. I will look into this.

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: Now that I think of it, nothing really needs to be handled in regards to the values. Simply adding the following line `update["options.$.values"] = values`, should do the work.

Comment (srindom) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: Does it even make sense to allow value updates? The values are stored on the ProductVariants so the values array would just be an aggregate of what is held in the product's variants. I think we should remove it altogether and let values live in the ProductVariant objects only

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/products/update-option.js:30]: I actually think, we spoke about this at some point in the past. I remember we chose to keep it for some reason, so maybe we should keep as it is for now, and then look at it at a later point? But I will remove the values update again, so only title can be touched.

File: packages/medusa/src/api/routes/store/products/__tests__/get-product.js [hunk 7]
```diff
@@ -1,93 +1,70 @@
 import mongoose from "mongoose"
 import getProduct from "../get-product"
+import { request } from "../../../../../helpers/test-request"
+import { IdMap } from "medusa-test-utils"
+import { ProductServiceMock } from "../../../../../services/__mocks__/product"
 
 describe("Get product by id", () => {
   const testId = `${mongoose.Types.ObjectId("56cb91bdc3464f14678934ca")}`
```
Comment (srindom) [inline packages/medusa/src/api/routes/store/products/__tests__/get-product.js:8]: Why not use `IdMap.getId("...")`?

File: packages/medusa/src/api/routes/store/products/__tests__/get-product.js [hunk 8]
```diff
@@ -1,93 +1,70 @@
...
+  // }
 
-  const resMock = {
-    sendStatus: jest.fn().mockReturnValue(),
-    json: jest.fn().mockReturnValue(),
-  }
+  // const resMock = {
+  //   sendStatus: jest.fn().mockReturnValue(),
+  //   json: jest.fn().mockReturnValue(),
+  // }
```
Comment (srindom) [inline packages/medusa/src/api/routes/store/products/__tests__/get-product.js:36]: Don't leave commented code :)

**Pull request merged**